lib/spack/spack/test/tag.py

```diff
@@ -0,0 +1,160 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+"""Tests for tag index cache files."""
+
+import pytest
+from six import StringIO
+
+import spack.cmd.install
+import spack.tag
+from spack.main import SpackCommand
+
+install = SpackCommand('install')
+
+# Alternate representation
+tags_json = \
+    """
+    {
+      "tags": {
+        "no-version": [
+          "noversion",
+          "noversion-bundle"
+        ],
+        "no-source": [
+          "nosource"
+        ]
+      }
+    }
+    """
+
+more_tags_json = \
+    """
+    {
+      "tags": {
+        "merge": [
+          "check"
+        ]
+      }
+    }
+    """
+
+
+def test_tag_copy(mock_packages):
+    index = spack.tag.TagIndex.from_json(StringIO(tags_json))
+    new_index = index.copy()
+
+    assert index.tags == new_index.tags
+
+
+def test_tag_get_all_available(mock_packages):
+    for skip in [False, True]:
+        all_pkgs = spack.tag.get_tag_packages([], False, skip)
```

Comment:
Minor. I know dynamic typing is not the best for these kind of things, but I would be consistent with the docstring:
```suggestion
        all_pkgs = spack.tag.get_tag_packages(None, False, skip)
```